tests/leak-tests/src/test/java/org/apache/activemq/artemis/tests/leak/ClientLeakTest.java

```diff
@@ -0,0 +1,214 @@
+         server.start();
+         System.out.println(LEAK_SERVER);
+      } catch (Throwable e) {
+         e.printStackTrace();
+         System.exit(-1);
+      }
+
+   }
+
+   @BeforeClass
+   public static void beforeClass() throws Exception {
+      Assume.assumeTrue(CheckLeak.isLoaded());
+   }
+
+   Process serverProcess;
+
+   @Override
+   @Before
+   public void setUp() throws Exception {
+      serverProcess = SpawnedVMSupport.spawnVM(ClientLeakTest.class.getName());
+      runAfter(serverProcess::destroyForcibly);
+
+      boolean success = false;
+      long time = System.currentTimeMillis() + 5_000;
+
+      // this loop will keep trying a connection until the serer has started
+      do {
+         try {
+            ConnectionFactory cf = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:61616");
+            try (Connection connection = cf.createConnection()) {
+               success = true;
+            }
+         } catch (Throwable e) {
+            logger.debug(e.getMessage(), e);
+            Thread.sleep(100);
+         }
+
+      } while (success == false && System.currentTimeMillis() < time);
+      Assert.assertTrue(success);
+   }
+
+   @After
+   public void stopServer() throws Exception {
+      serverProcess.destroyForcibly();
+   }
+
+   @Test
+   public void testAMQP() throws Exception {
+      final String protocol = "AMQP";
+      CheckLeak checkLeak = new CheckLeak();
+      int REPEATS = 100;
+      int MESSAGES = 20;
+      basicMemoryAsserts();
+
+      ConnectionFactory cf = CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616");
+
+      try (Connection producerConnection = cf.createConnection(); Connection consumerConnection = cf.createConnection()) {
+
+         Session producerSession = producerConnection.createSession(true, Session.SESSION_TRANSACTED);
+
+         Session consumerSession = consumerConnection.createSession(true, Session.SESSION_TRANSACTED);
+         consumerConnection.start();
+         assertMemory(checkLeak, 0, ReceiverImpl.class.getName());
+
+         for (int i = 0; i < REPEATS; i++) {
+            System.out.println("********************************************************************************");
+            {
+               Destination source = producerSession.createQueue("source");
+               try (MessageProducer sourceProducer = producerSession.createProducer(source)) {
+                  for (int msg = 0; msg < MESSAGES; msg++) {
+                     Message message = producerSession.createTextMessage("hello " + msg);
+                     message.setIntProperty("i", msg);
+                     sourceProducer.send(message);
+                  }
+                  sourceProducer.close();
+                  producerSession.commit();
+               }
+            }
+
+            // it shouldn't have any sender connects still. The retry loop is leaving a few behind.:w
+            assertMemory(checkLeak, 5, SenderImpl.class.getName());
```

Comment:
@gemmellr SenderImpl keeps piling up on this test... Even though the producers and consumers are  closed... 

I think this is similar to what I see on the server's.

Reply:
false alarm... I had a producer I forgot to close. duh